herddb-core/src/main/java/herddb/core/TableSpaceManager.java

```diff
@@ -834,6 +838,37 @@ void metadataUpdated(TableSpace tableSpace) {
         dataStorageManager.tableSpaceMetadataUpdated(tableSpace.uuid, tableSpace.expectedReplicaCount);
     }
 
+    AbstractTableManager getTableManagerByUUID(String parentTableId) {
+        //TODO: make this method more efficient
+        for (AbstractTableManager manager : tables.values()) {
+            if (manager.getTable().uuid.equals(parentTableId)) {
+                return manager;
+            }
+        }
+        throw new HerdDBInternalException("Cannot find tablemanager for " + parentTableId);
+    }
+
+    public Table[] collectChildrenTables(Table childTable) {
```

Comment:
Parameter childTable shouldn't be "parent"?

Reply:
parameter renamed to parentTable